python/shotgun_fields/shotgun_field_manager.py

```diff
@@ -107,39 +22,101 @@ class ShotgunFieldManager(QtCore.QObject):
     for the set of widgets that can display values from Shotgun in a way appropriate
     to their field type.
 
-    :signal initialized(): Fires when the manager has finished running all the background tasks
+    :signals:
+        ``initialized()`` - Fires when the manager has finished running all the background tasks
         it needs for its functionality
+
+    :enum: ``DISPLAY, EDITOR, EDITABLE`` - Enumeration for widget types managed and
+        provided by the class
     """
 
     # dictionary that keeps the mapping from Shotgun data type to widget class
-    __FIELD_TYPE_MAP = {}
+    __WIDGET_TYPE_CLS_MAP = {}
 
-    # first when we are ready to manage the widgets
+    # fires when we are ready to manage the widgets
     initialized = QtCore.Signal()
 
+    # widget types enumeration
+    _WIDGET_TYPES = (DISPLAY, EDITOR, EDITABLE) = ("display", "editor", "editable")
+
+    ############################################################################
+    # class methods
+
+    @classmethod
+    def get_class(cls, sg_entity_type, field_name, widget_type=DISPLAY):
```

Comment:
i thought that initially too. but this method is part of the `ShotgunFieldManager` definition so that namespace isn't available yet. but the widget type attributes have been defined (they've above this line) so this seems to work. Not sure if this is good practice though. not sure I can remember doing this before.